src/main/java/com/google/impactdashboard/database_manager/data_read/DataReadManagerImpl.java

```diff
@@ -114,6 +114,27 @@ public double getAverageIAMBindingsInPastYear(String projectId) {
     return datesToBindings;
   }
 
+  /**
+   * Queries the IAM Table and returns the most recent timestamp, or -1 if there is
+   * no data.
+   */
+  public long getMostRecentTimestamp() {
+    QueryJobConfiguration queryConfiguration = queryConfigurationBuilder
+      .getMostRecentTimestampConfiguration()
+      .build();
+    TableResult results = database.readDatabase(queryConfiguration);
```

Comment:
The query seems should only return one row? Consider use Iterables.getOnlyElement instead of a loop.

Reply:
!!!! Thank you so much for pointing this function out to me!! This is so helpful! I'm going to update all of the rest of my code to use this now!